### ISSUE TITLE
Unable to login to webmail

### ISSUE THREAD

**Issue by syllith** [modoboa/modoboa-webmail#175]
Impacted versions: 1.10.6
Distribution: Debian
Codename: Stretch
Arch: 64 Bits
Database: MySQL

#Steps to reproduce:
I did a normal installation of Modoboa with no problem. Was able to login to the web panel and create my users. Was able to log in to new users account by when clicking on webmail I received : "Error: command: LOGIN => Internal error occurred. Refer to server log for more information."

#Full trace using --debug option or current behavior:
I attempted to view log files but for the life of me cannot find where it is stored. I tried the one built into the Modoboa web client but nothing. What logs I could find on the server revealed nothing except maybe the one in /var/log/mail.log which told me repeatedly that amavis could not open lock file. I'm unsure if this is related to this issue but it's the only thing I could find. If you have any locations youd like me to check, I'll be more than happy to share them.

#Expected behavior:
I would love to be able to use a client such as rainloop to log in to my accounts I create using the modoboa web client but as far as I'm concerned right now I would like to be able to just access the webmail without error. (Also it appears the ssl cert is invalid for the page but it lets me access it anyway, unsure if that could be affecting authentication somehow)

Thank you for any help. I'm at a loss here.

**Comment by syllith** [comment]
Sure, I'm get this error repeatedly:

Sep  5 02:50:30 endurehosting amavis[29442]: (!)Net::Server: 2018/09/05-02:50:30 Couldn't open lock file "/var/run/amavis/amavisd.lock"[Permission denied]\n  at line 177 in file /usr/share/perl5/Net/Server/PreForkSimple.pm

**Comment by tonioo** [comment]
Make sure this file belongs to the amavis system user and has write permission enabled.

**Comment by syllith** [comment]
I have just ensured it was and it has full read and write access. Still getting the same thing. Would this be the reason I cannot get through to webmail?

**Comment by tonioo** [comment]
Maybe... Have you restarted amavis after you changed file's permissions?

**Comment by syllith** [comment]
I restarted the entire server

**Comment by syllith** [comment]
Sorry to keep bugging you, but I'm still having the same issue. Kind of desperate to be honest. Haven't been able to get emails in almost a week now. I've reinstalled modoboa and have been comparing the settings of the online demo with mine and I've tried a few combinations of things but I'm still having the same error. Funny though, it looks like I've been capable of receiving emails since I set it up. If I got to statistics, it looks like I've been getting them because it's been either rejecting them due to spam and some have been accepted too. I just can't understand why it won't authenticate. I've played around with some amavis configs and permissions and I'm not seeing the permission denied error anymore. In fact I cleared my mail.log and I haven't had anything so either it's not logging or we're all good on that. When I get that error when logging into the webmail and it tells me to check the log, what log is it referring to exactly? And thank you for any further help you may provide. I've been checking out some other posts and you're good at keeping up with people. Appreciate that.

**Comment by tonioo** [comment]
@syllith Can you paste your logs somewhere. There should be something inside after an unsuccessful login attempt...

**Comment by syllith** [comment]
Hey man, sorry for the late response. I got everything working 100%. Modoboa truly is amazing. So for future reference for anybody having issues with it, modoboa works great. The problem was because I had previously attempted to install and configure my mail server manually using dovecot and postfix. I must've messed with so many files that when modoboa went to install, it must've glossed over a few things. I ended up reinstalling debian from scratch and installing modoboa first thing. Tried it, and the web mail was working fine. Was even able to send / receive emails. Unfortunately, I was unable to use rainloop to log in, which is what I was wanting to use. Set imap security to starttls and port 143, then smtp to port 587, also starttls. Could not select short login however, but thats fine with me. Did not have to use php mail() function which I had to use on my windows server using hmail, so I guess thats a plus.

This allowed me to log in but I couldn't send emails and could not open received ones. I'd just get "unknown error" when trying to open them. Found a thread online and a guy just reinstalled rainloop and it worked fine. The rainloop installer told me I was missing php-dom and php-curl which could have been why. Now everything is working great. 

I was really considering switching back to windows because this webmail thing was really stressing me out. Thanks again for all your help. Didn't mean to waste your time but I didn't know where else to turn. Keep it up my man.
